internal/components/compute/azcommands.go

```diff
@@ -1,57 +1,238 @@
 package compute
 
 import (
-	"github.com/Azure/aks-mcp/internal/utils"
+	"fmt"
+	"slices"
+
+	"github.com/Azure/aks-mcp/internal/config"
 	"github.com/mark3labs/mcp-go/mcp"
 )
 
-// ComputeCommand defines a specific az vmss command to be registered as a tool
-type ComputeCommand struct {
-	Name        string
-	Description string
-	ArgsExample string // Example of command arguments
-}
+// ComputeOperationType defines the type of compute operation
+type ComputeOperationType string
+
+// ResourceType defines the compute resource type
+type ResourceType string
+
+const (
+	// VM operations
+	OpVMShow            ComputeOperationType = "show"
+	OpVMList            ComputeOperationType = "list"
+	OpVMCreate          ComputeOperationType = "create"
+	OpVMDelete          ComputeOperationType = "delete"
+	OpVMStart           ComputeOperationType = "start"
+	OpVMStop            ComputeOperationType = "stop"
+	OpVMRestart         ComputeOperationType = "restart"
+	OpVMDeallocate      ComputeOperationType = "deallocate"
+	OpVMResize          ComputeOperationType = "resize"
```

Comment:
Could we only keep read+restart/reimage/run-command and get other write/delete operations removed?

Reply:
creating a new VM is complidated, without querying the available vm size, quota in required regions, we don't actually have a good way to let AI to create a VM with high success rate.

Reply:
ok. removed those operations.